pinot-filesystem/src/main/java/com/linkedin/pinot/filesystem/LocalPinotFS.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.filesystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.io.FileUtils;
+
+
+/**
+ * Implementation of PinotFS for a local filesystem. Methods in this class may throw a SecurityException at runtime
+ * if access to the file is denied.
+ */
+public class LocalPinotFS extends PinotFS {
+
+  public LocalPinotFS() {
+  }
+
+  @Override
+  public void init(Configuration configuration) {
+  }
+
+  @Override
+  public boolean delete(URI segmentUri) throws IOException {
+    File file = new File(segmentUri);
+    if (file.isDirectory()) {
+      // Throws an IOException if it is unable to delete
+      FileUtils.deleteDirectory(file);
+    } else {
+      // Returns false if delete fails
+      return FileUtils.deleteQuietly(file);
+    }
+    return true;
+  }
+
+  @Override
+  public boolean move(URI srcUri, URI dstUri) throws IOException {
+    File srcFile = new File(srcUri);
+    File dstFile = new File(dstUri);
+    if (dstFile.exists()) {
+      FileUtils.forceDelete(dstFile);
+    }
+    if (!srcFile.isDirectory() && !dstFile.exists()) {
+      dstFile.getParentFile().mkdirs();
+      dstFile.createNewFile();
+    }
+    if (srcFile.isDirectory() && !dstFile.exists()) {
+      dstFile.mkdirs();
+    }
+    FileUtils.moveFile(srcFile, dstFile);
+    return true;
+  }
+
+  @Override
+  public boolean copy(URI srcUri, URI dstUri) throws IOException {
+    File srcFile = new File(srcUri);
+    File dstFile = new File(dstUri);
+    if (srcFile.isDirectory()) {
+      // Throws Exception on failure
+      FileUtils.copyDirectory(srcFile, dstFile);
+    } else {
+      // Will create parent directories, throws Exception on failure
+      FileUtils.copyFile(srcFile, dstFile);
+    }
+    return true;
+  }
+
+  @Override
+  public boolean exists(URI segmentUri) throws IOException {
+    File file = new File(segmentUri);
+    return file.exists();
+  }
+
+  @Override
+  public long length(URI segmentUri) throws IOException {
+    File file = new File(segmentUri);
+    return FileUtils.sizeOf(file);
+  }
+
+  @Override
+  public String[] listFiles(URI segmentUri) throws IOException {
+    File file = new File(segmentUri);
+    Collection<File> files = FileUtils.listFiles(file, null, true);
```

Comment:
The interface states this should actually return a null (or fix the interface def to reflect the right thing) - ensure its consistent and there is a test case that asserts for this.